go/saltpack/encrypt_test.go

```diff
@@ -763,7 +763,7 @@ func TestCorruptEncryption(t *testing.T) {
 		t.Fatal(err)
 	}
 	_, _, err = Open(ciphertext, kr)
-	if mm, ok := err.(ErrBadCiphertext); !ok {
```

Comment:
Note that in this approach, it's impossible to produce `ErrBadCiphertext` by corrupting the ciphertext after encryption, because the per-recipient authenticator now covers the entire ciphertext and always leads to an `ErrBadTag`. To get this error, we have to corrupt the ciphertext before the authenticators are computed by the sender. Line 707 tests this by messing with the session key.